robot/EDA/resources/EDA.py

```diff
@@ -381,3 +424,187 @@ def get_namespace_prefix(self, name):
+        locator_menu = eda_lex_locators["navigation_menu"]
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        locator_tab = eda_lex_locators["navigation_tab"].format(tab)
+
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Navigation menu drop down unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+        self.selenium.capture_page_screenshot()
+        self.selenium.wait_until_page_contains_element(
+            locator_tab,
+            error=tab + " item not found as an available option in: " + locator_tab
+        )
+        self.selenium.click_element(locator_tab)
+
+
+
+    def select_tab(self, tab):
+        """ Selects Contacts tab - as passed in to the function """
+        locator_menu = eda_lex_locators["tab_menu"].format(tab)
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        locator_tab = eda_lex_locators["tab_tab"].format(tab)
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Contact tab unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+
+    def select_value_from_listbox(self, title, value):
+        """ Selects value from a listbox/dropdown identified by title """
+        locator_title = eda_lex_locators["listbox"]["title"].format(title)
+        locator_value = eda_lex_locators["listbox"]["value"].format(value)
+        self.selenium.click_element(locator_title)
+        time.sleep(1)
+        self.selenium.click_element(locator_value)
+
+    def select_xpath(self, loc, value):
+        """ Selects the correct xpath by checking if it exists on the page
+            from the given list of locator possibilities
+        """
+        locators = eda_lex_locators[loc].values()
+
+        for i in locators:
+            locator = self.format_all(i, value)
+            if self.check_if_element_exists(locator):
+                return locator
+
+        assert "Button with the provided locator not found"
```

Comment:
This isn't the correct usage of `assert`. The first argument to `assert` is a condition. When you pass a string, it is treated as a condition that returns `True` (as in, it's true that the string has characters)

The correct way is to raise an exception: `raise Exception("Button with the provided locator not found")`. Even better would be to include the locator value in the error.

Reply:
This is actually someone else's code from a long time ago.  I'd prefer to not touch this code as it is not even being used for this work item.

Reply:
This definitely needs to be fixed. It's not doing what you think it's doing - it will not raise an error. The keyword will appear to have succeeded. 

However, now that I do a bit more investigation, I don't see that it's being used anywhere. You should delete it.

Reply:
I removed this broken function